lib/ip_list_write.go

```diff
@@ -161,54 +160,105 @@ func IPListWriteFromIPRangeStr(rStr string) error {
 
 // IPListWriteFromReader returns a list of IPs after reading from a reader; the
 // reader should have IPs per-line.
-func IPListWriteFromReader(r io.Reader) {
+func IPListWriteAllFromReader(r io.Reader) {
+	IPListWriteFromReader(r, true, true, true)
+}
+
+// IPListWriteFromReader returns a list of IPs after reading from a reader
+// from selected sources; the reader should have IPs per-line.
+func IPListWriteFromReader(
+	r io.Reader,
+	ip bool,
+	iprange bool,
+	cidr bool) {
 	scanner := bufio.NewScanner(r)
 	for scanner.Scan() {
 		ipStr := strings.TrimSpace(scanner.Text())
```

Comment:
This is no longer necessarily an "ip string" so the name `ipStr` is inappropriate; you can use `input` instead,  just as the function `IPListWriteFrom` does.